hpOneView/resources/resource.py

```diff
@@ -502,6 +575,91 @@ def get_members(self, mlist):
         else:
             return []
 
+    def do_requests_to_getall(self, uri, requested_count):
+        """Helps to make http request for get_all method.
+
+        Note:
+            This method will be checking for the pagination URI in the response
+            and make request to pagination URI to get all the resources.
+        """
+        items = []
+
+        while uri:
+            logger.debug('Making HTTP request to get all resources. Uri: {0}'.format(uri))
+            response = self._connection.get(uri)
+            members = self.get_members(response)
+            items += members
+
+            logger.debug("Response getAll: nextPageUri = {0}, members list length: {1}".format(uri, str(len(members))))
+            uri = self.get_next_page(response, items, requested_count)
+
+        logger.debug('Total # of members found = {0}'.format(str(len(items))))
+        return items
+
+    def get_next_page(self, response, items, requested_count):
+        """Returns next page URI."""
+        next_page_is_empty = response.get('nextPageUri') is None
+        has_different_next_page = not response.get('uri') == response.get('nextPageUri')
+        has_next_page = not next_page_is_empty and has_different_next_page
+
+        if len(items) >= requested_count and requested_count != -1:
+            return None
+
+        return response.get('nextPageUri') if has_next_page else None
+
+    def do_get(self, uri):
+        """Helps to make get requests
+
+        Args:
+            uri: URI of the resource
+
+        Returns:
+            Returns: Returns the resource data
+        """
+        self.validate_resource_uri(uri)
```

Comment:
Same

Reply:
validate_resource_uri is a generic method to check the resource base URL. its used by all the main REST calls to make sure the resource URI is valid. 

We should throw an exception if the resource base URI is not valid and break the execution and let the user know there is an issue. If we are catching this exception it will be difficult for the user to debug a serious issue.